### DISCUSSION TITLE
big difference between IC1 voltage and IC2

### DISCUSSION THREAD

**Discussion by p3ck** [CircuitSetup/Expandable-6-Channel-ESP32-Energy-Meter#114] [Q&A] [answered]
I'm clearly doing something wrong.

I'm currently just experimenting with a split extension cord and a kill-a-watt to make sure I'm reading accurately.  I have 2 100 amp clamps that do show .44 amps when a 60 watt lightbulb is plugged in. (matched the kill-a-watt reading).

But I get a crazy 655 volt reading on IC1, IC2 shows 114

![image](https://user-images.githubusercontent.com/2686136/196747874-2eaa7142-93ab-49fd-bd36-39fd712f2e50.png)

I have not cut any traces on the 6 channel board,  I understand that if I want accurate readings from the split phase I would need to power via the second transformer which would be on the other phase.  But I didn't think this would be that off?

realistically I am thinking I will just use the voltage value from IC2 and assume my other phase is close to that..

Here is my config:

`substitutions:
# Change the disp_name to something you want  
  disp_name: 6C
# Interval of how often the power is updated
  update_time: 10s
# Current Transformers:
#  20A/25mA SCT-006: 11143
#  30A/1V SCT-013-030: 8650
#  50A/1V SCT-013-050: 15420
#  80A/26.6mA SCT-010: 41660
#  100A/50ma SCT-013-000: 27518
#  120A/40mA: SCT-016: 41787
#  200A/100mA SCT-024: 27518
  current_cal: '27518'
# Jameco 9VAC Transformer: 42620
  voltage_cal: '42620'

esphome:
  name: "esphome-bsmnt-9c8adc"

esp32:
  board: esp32dev
  framework:
    type: arduino

# mqtt:
#  broker: !secret mqtt_broker
#  username: !secret mqtt_user
#  password: !secret mqtt_pass

# Enable logging
logger:

# Enable Home Assistant API
api:

ota:

web_server:
  port: 80

spi:
  clk_pin: 18
  miso_pin: 19
  mosi_pin: 23

sensor:
  - platform: wifi_signal
    name: ${disp_name} WiFi
    update_interval: 60s
#IC1
  - platform: atm90e32
    cs_pin: 5
    phase_a:
      voltage:
        name: ${disp_name} Volts A
        id: ic1Volts
        accuracy_decimals: 1
      current:
        name: ${disp_name} CT1 Amps
        id: ct1Amps
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    frequency:
      name: ${disp_name} Freq A
    line_frequency: 60Hz
    gain_pga: 1X
    update_interval: ${update_time}
#IC2
  - platform: atm90e32
    cs_pin: 4
    phase_c:
      voltage:
        name: ${disp_name} Volts B
        id: ic2Volts
        accuracy_decimals: 1
      current:
        name: ${disp_name} CT6 Amps
        id: ct6Amps
      gain_ct: ${current_cal}
    frequency:
      name: ${disp_name} Freq B
    line_frequency: 60Hz
    gain_pga: 1X
    update_interval: ${update_time}
    
#Watts per channel
  - platform: template
    name: ${disp_name} CT1 Watts
    id: ct1Watts
    lambda: return id(ct1Amps).state * id(ic1Volts).state;
    accuracy_decimals: 0
    unit_of_measurement: W
    device_class: power
    update_interval: ${update_time}         
  - platform: template
    name: ${disp_name} CT6 Watts
    id: ct6Watts
    lambda: return id(ct6Amps).state * id(ic2Volts).state;
    accuracy_decimals: 0
    unit_of_measurement: W
    device_class: power
    update_interval: ${update_time}         
#Total Amps   
  - platform: template
    name: ${disp_name} Total Amps
    id: totalAmps
    lambda: return id(ct1Amps).state + id(ct6Amps).state ;
    accuracy_decimals: 2
    unit_of_measurement: A
    device_class: current
    update_interval: ${update_time}         
#Total Watts
  - platform: template
    name: ${disp_name} Total Watts
    id: totalWatts
    lambda: return id(totalAmps).state * id(ic1Volts).state;
    accuracy_decimals: 1
    unit_of_measurement: W
    device_class: power
    update_interval: ${update_time}
#kWh
  - platform: total_daily_energy
    name: ${disp_name} Total kWh
    power_id: totalWatts
    filters:
      - multiply: 0.001
    unit_of_measurement: kWh
    device_class: energy
    state_class: total_increasing
switch:
  - platform: restart
    name: ${disp_name} Restart  
time:
  - platform: sntp
    id: sntp_time   

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password
    `

**Comment by p3ck** [comment, marked as answer]
I figured this out with a little help on the discord channel.

I had the wrong voltage_cal for my revision of the 6 channel board.  I changed from voltage_cal: '42620' to voltage_cal: '7305' and that fixed it.  So pay attention to the examle configs and the version of the board you have.

**Comment by CircuitSetup** [comment]
Thanks for following up! I'm glad you were able to figure it out!
